src/Uno/Prism.Uno/PrismApplicationBase.cs

```diff
@@ -0,0 +1,226 @@
+            containerRegistry.Register<IRegionNavigationJournal, RegionNavigationJournal>();
+            containerRegistry.Register<IRegionNavigationService, RegionNavigationService>();
+        }
+
+        /// <summary>
+        /// Used to register types with the container that will be used by your application.
+        /// </summary>
+        protected abstract void RegisterTypes(IContainerRegistry containerRegistry);
+
+        /// <summary>
+        /// Configures the <see cref="IRegionBehaviorFactory"/>. 
+        /// This will be the list of default behaviors that will be added to a region. 
+        /// </summary>
+        protected virtual void ConfigureDefaultRegionBehaviors(IRegionBehaviorFactory regionBehaviors)
+        {
+            if (regionBehaviors != null)
+            {
+                regionBehaviors.AddIfMissing(BindRegionContextToDependencyObjectBehavior.BehaviorKey, typeof(BindRegionContextToDependencyObjectBehavior));
+                regionBehaviors.AddIfMissing(RegionActiveAwareBehavior.BehaviorKey, typeof(RegionActiveAwareBehavior));
+                regionBehaviors.AddIfMissing(SyncRegionContextWithHostBehavior.BehaviorKey, typeof(SyncRegionContextWithHostBehavior));
+                regionBehaviors.AddIfMissing(RegionManagerRegistrationBehavior.BehaviorKey, typeof(RegionManagerRegistrationBehavior));
+                regionBehaviors.AddIfMissing(RegionMemberLifetimeBehavior.BehaviorKey, typeof(RegionMemberLifetimeBehavior));
+                regionBehaviors.AddIfMissing(ClearChildViewsRegionBehavior.BehaviorKey, typeof(ClearChildViewsRegionBehavior));
+                regionBehaviors.AddIfMissing(AutoPopulateRegionBehavior.BehaviorKey, typeof(AutoPopulateRegionBehavior));
+                regionBehaviors.AddIfMissing(IDestructibleRegionBehavior.BehaviorKey, typeof(IDestructibleRegionBehavior));
+            }
+        }
+
+        /// <summary>
+        /// Configures the default region adapter mappings to use in the application, in order
+        /// to adapt UI controls defined in XAML to use a region and register it automatically.
+        /// May be overwritten in a derived class to add specific mappings required by the application.
+        /// </summary>
+        /// <returns>The <see cref="RegionAdapterMappings"/> instance containing all the mappings.</returns>
+        protected virtual void ConfigureRegionAdapterMappings(RegionAdapterMappings regionAdapterMappings)
+        {
+            if (regionAdapterMappings != null)
+            {
+                regionAdapterMappings.RegisterMapping(typeof(Selector), _containerExtension.Resolve<SelectorRegionAdapter>());
+                regionAdapterMappings.RegisterMapping(typeof(ItemsControl), _containerExtension.Resolve<ItemsControlRegionAdapter>());
+                regionAdapterMappings.RegisterMapping(typeof(ContentControl), _containerExtension.Resolve<ContentControlRegionAdapter>());
+            }
```

Comment:
let's update this to use

```cs
regionAdapterMappings.RegisterMapping<Selector, SelectorRegionAdapter>();
```

Reply:
I could not replace it with this line (it causes a wire-up exception), but @brianlagunas made a common method for `ApplicationBase`, so I made the change to use this instead:

https://github.com/PrismLibrary/Prism/blob/4a109d697eeeeef3274bff1c936c186616c6bf71/src/Wpf/Prism.Wpf/PrismInitializationExtensions.cs#L55

Reply:
why isn't this using the extension methods like the others?